src/util.js

```diff
@@ -107,3 +107,80 @@ export function format_memory_and_limit(usage, limit) {
         return _("");
     }
 }
+
+export function updateContainers() {
+    let newContainers = {};
+    let newContainersStats = {};
+    let newContainersMeta = [];
+    let inspectRet = [];
+    let containerStatsRet = [];
+    let countRun;
+    return new Promise((resolve, reject) => {
+        varlinkCall(PODMAN, "io.podman.ListContainers")
+                .then(reply => {
+                    newContainersMeta = reply.containers;
+                    if (!newContainersMeta) {
+                        console.log(newContainersMeta);
+                        resolve({newContainers: newContainers, newContainersStats: newContainersStats});
```

Comment:
Note that this does *not* magically end the handler function, i. e. this is not an implied `return`. All of the code below will still run and resolve the promise again, with (hopefully :smile: ) the same result. I suggest to just drop this `if` condition - everything above should get along with the list being empty.